src/TimingAnalysis.cc

```diff
@@ -39,216 +37,274 @@ double sgn(double val){
+    for (int i = 0; i < _pythiaPU->event.size(); ++i) {
+
+      //skipping Leptons
+      if (!_pythiaPU->event[i].isFinal()    ) continue;
+      if (fabs(_pythiaPU->event[i].id())==12) continue;
+      if (fabs(_pythiaPU->event[i].id())==14) continue;
+      if (fabs(_pythiaPU->event[i].id())==13) continue;
+      if (fabs(_pythiaPU->event[i].id())==16) continue;
       
-      if (!pythia8->event[ip].isFinal() )      continue;
-      //if (fabs(pythia8->event[ip].id())  ==11) continue;
-      if (fabs(pythia8->event[ip].id())  ==12) continue;
-      if (fabs(pythia8->event[ip].id())  ==13) continue;
-      if (fabs(pythia8->event[ip].id())  ==14) continue;
-      if (fabs(pythia8->event[ip].id())  ==16) continue;
+      //Instantiate new pseudojet
+      PseudoJet p(_pythiaPU->event[i].px(), _pythiaPU->event[i].py(), _pythiaPU->event[i].pz(),_pythiaPU->event[i].e() ); 
       
-      fastjet::PseudoJet p(pythia8->event[ip].px(), pythia8->event[ip].py(), pythia8->event[ip].pz(),pythia8->event[ip].e() ); 
+      //extract event information
       double eta = p.rapidity();
-      if (fabs(eta)>5.0) continue;
-      double corrtime = 0.;
-      if (fabs(eta)<minEta) corrtime = -999.;
-      p.reset_PtYPhiM(p.pt(), eta, p.phi(), 0.);
-      p.set_user_info(new TimingInfo(pythia8->event[ip].id(),ip,0, false,corrtime)); //0 for the primary vertex. 
+      double sinheta = sinh(eta);
+      double cosheta = cosh(eta);
       
-      particlesForJets.push_back(p);
-      particlesForJets_np.push_back(p);
-
-    } // end particle loop -----------------------------------------------
-
-    fastjet::JetDefinition jetDef(fastjet::antikt_algorithm, 0.4, fastjet::E_scheme, fastjet::Best);
-    fastjet::AreaDefinition active_area(fastjet::active_area);
-    fastjet::ClusterSequenceArea clustSeq(particlesForJets, jetDef, active_area);
-
-    fastjet::GridMedianBackgroundEstimator bge(4.5,0.6);
-    bge.set_particles(particlesForJets);
-    fastjet::Subtractor subtractor(&bge);
-
-    vector<fastjet::PseudoJet> inclusiveJets = sorted_by_pt(clustSeq.inclusive_jets(10.));
-    vector<fastjet::PseudoJet> subtractedJets = subtractor(inclusiveJets);
-    Selector select_fwd = SelectorAbsRapRange(minEta,4.5);
-    vector<fastjet::PseudoJet> selectedJets = select_fwd(subtractedJets);
-
-    FillTree(selectedJets);
-
-    fastjet::ClusterSequenceArea clustSeqTruth(particlesForJets_np, jetDef, active_area);
-    vector<fastjet::PseudoJet> truthJets = sorted_by_pt(clustSeqTruth.inclusive_jets(10.));
-    vector<fastjet::PseudoJet> selectedTruthJets = select_fwd(truthJets);
-    FillTruthTree(selectedTruthJets);
-
-    tT->Fill();
-
-    if(fDebug) 
-      cout << "TimingAnalysis::AnalyzeEvent End " << endl;
-
-    return;
+      //calculate eta from displacement (minEta pos)
+      static const double radius = 1.2; // barrel radius=1.2 meter
+      double zbase = radius*sinh(minEta)*sgn(eta)-zhs; //displace due to new location of Hard-Scatter Vertex
```

Comment:
I am not sure this is what we want. The actual Eta of pileup particles does not depend on zhs, but only on the position of the pileup vertex.
It's only the reference time "reftime" that should take into account the position of the hs vertex.